relay-sampling/src/evaluation.rs

```diff
@@ -93,6 +207,9 @@ impl SamplingEvaluator {
                         self.rule_ids,
                     ));
                 }
+                SamplingValue::Reservoir { .. } => {
+                    return ControlFlow::Break(SamplingMatch::new(1.0, seed, vec![rule.id]));
```

Comment:
we only take the last rule id in the reservoir case because the reservoir matching overrides all previously matched rules.